landingai/predict.py

```diff
@@ -750,13 +756,24 @@ def _do_inference(
     data: Optional[Dict[str, Any]] = None,
 ) -> List[Prediction]:
     """Call the inference endpoint and extract the prediction result."""
-    try:
-        resp = session.post(endpoint_url, files=files, params=payload, data=data)
-    except requests.exceptions.ConnectionError as e:
-        raise ConnectionError(
-            f"Failed to connect to the model server. Please double check the model server url ({endpoint_url}) is correct.\nException detail: {e}"
+
+    # Make number of requests required
+    while True:
+        try:
+            resp = session.post(endpoint_url, files=files, params=payload, data=data)
+        except requests.exceptions.ConnectionError as e:
+            raise ConnectionError(
+                f"Failed to connect to the model server. Please double check the model server url ({endpoint_url}) is correct.\nException detail: {e}"
+            )
+        response = HttpResponse.from_response(resp)
+        if response.status_code != 429:
+            # The requests was successful or failed not due to the rate limiter
+            break
+        _LOGGER.warning(
+            "Got HTTP 429: You are exceeding your cloud inference rate. Sleeping for 60 sec"
         )
-    response = HttpResponse.from_response(resp)
+        time.sleep(60)
+
```

Comment:
sense a little code smell with this "while true" loop. this logic combined with the existing Retry makes debugging and testing harder...   (is it worth it?)

can we tune the existing `backoff_factor` and other parameters in Retry instead? i.e. to let the Retry wait for a minute or half a minute

Reply:
No. I tried to looking into urllib and request for tailored retry policies per error code...I even considered implementing a custom Retry class but it is not really designed to be extended.

About the `while True`, i do not like it but I could not think of an exit condition. It is ok for a customer that has 40 req/min to wait for a day to complete a video.... note that an exception like http 500 will break the look...

Reply:
for a custom retry solution, you can explore the [Tenacity](https://tenacity.readthedocs.io/en/latest/) library if you feel like trying something new and have more time (our existing tests may need some tweaks).

Reply:
Done. Used tenacity